Source/Core/VideoCommon/AbstractTexture.h

```diff
@@ -0,0 +1,34 @@
+// Copyright 2009 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#pragma once
+
+#include "Common/CommonTypes.h"
+#include "Common/MathUtil.h"
+#include "VideoCommon/TextureConfig.h"
+#include "VideoCommon/VideoCommon.h"
+
+class AbstractTexture
+{
+public:
+  explicit AbstractTexture(const TextureConfig& c)
+    : config(c)
+  {}
+
+  virtual ~AbstractTexture()
+  {}
+
+  virtual void Bind(unsigned int stage) = 0;
```

Comment:
I'm not sure if Bind() should be a function of the texture object. It is more a state of draw calls.